src/main/java/org/mockito/internal/stubbing/answers/AnswerFunctionalInterfaces.java

```diff
@@ -0,0 +1,326 @@
+package org.mockito.internal.stubbing.answers;
+
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+/**
+ * Functional interfaces to make it easy to implement answers in Java 8
+ */
+public class AnswerFunctionalInterfaces {
+    /**
```

Comment:
Easily added if that's the house style. I find it unnecessarily drops the code coverage - who unit tests an unreachable private constructor? 

All those in favour of adding it?